src/auth/channel.ts

```diff
@@ -0,0 +1,73 @@
+import * as did from "../did/index.js"
+import { setup } from "../setup/internal.js"
+import { LinkingError } from "./linking.js"
+
+import type { Maybe } from "../common/index.js"
+
+export type Channel = {
+  send: (data: ChannelData) => void
+  close: () => void
+}
+
+export type ChannelOptions = {
+  username: string
+  handleMessage: (event: MessageEvent) => void
+}
+
+type ChannelData = string | ArrayBufferLike | Blob | ArrayBufferView
+
+export const createChannel = async (options: ChannelOptions): Promise<Channel> => {
+  const { username, handleMessage } = options
+
+  const rootDid = await await did.root(username).catch(_ => null)
```

Comment:
I know it's annoying that javascript (or rather eslint) doesn't recognize `_` as unused variable, but you can silence this warning by using `() => null` instead.

Reply:
Good point. I've updated this to `()`.